src/shared/bg_public.h

```diff
@@ -1525,6 +1525,159 @@ meansOfDeath_t            BG_MeansOfDeathByName( const char *name );
 void                      BG_InitAllConfigs();
 void                      BG_UnloadAllConfigs();
 
+/*
+ * This class is a simpler std::vector alternative that doesn't allocate
+ * and survives a memset(0) without damage, provided that you don't need to
+ * call destructors in such case.
+ *
+ * The purpose of this is to provide something in-between std::vector, which
+ * has fully dynamic and has unbounded size, and std::array that quite
+ * impractical as soon as you don't use it at its full capacity (it doesn't
+ * support an .append() or a clear() method, you can't use a C++11 style-loop
+ * to iterate over only the defined elements, etc.)
+ */
+template<typename T, size_t maximum_size_>
+class BoundedVector
+{
+private:
+	T array[maximum_size_];
+	size_t count_;
+public:
+	BoundedVector() : count_(0) {}
+	BoundedVector(const BoundedVector<T, maximum_size_>& other)
+		: count_(other.count_)
+	{
+		for (size_t i = 0; i < other.count_; i++) {
+			array[i] = other.array[i];
+		}
+	}
+	BoundedVector<T, maximum_size_>& operator=(const BoundedVector<T, maximum_size_>& other) {
+		for (size_t i = 0; i < other.count_; i++) {
+			array[i] = other.array[i];
+		}
+		count_ = other.count_;
+		return *this;
+	}
+	~BoundedVector() {
+		while (pop_back());
+	}
+
+	class iterator
+	{
+	private:
+		BoundedVector<T, maximum_size_>* vector_;
```

Comment:
You can do without an iterator class, by having the iterator be a `T*` pointing to the element.

Reply:
Neat!